config.go

```diff
@@ -394,9 +408,32 @@ func (cfg *Config) unshadow() {
 	cfg.shadow = nil
 }
 
+func generateSwarmSecret() (string, error) {
+	key := make([]byte, 32)
+	_, err := crand.Read(key)
+	if err != nil {
+		return "", fmt.Errorf("Error reading from rand: %v", err)
+	}
+	return hex.EncodeToString(key), nil
+}
+
+func swarmSecretToKey(secret string) (string, error) {
+	hexSecret := hex.EncodeToString([]byte(secret))[:68]
```

Comment:
@hsanjuan Thanks for catching the 68, that was purely for testing purposes. And @Kubuxu yeah that would be good I think, I'll make a PR soon if no one else gets to it first.